src/app/Processor/ProcessorDetailPage/ProcessorDetailPage.tsx

```diff
@@ -104,32 +124,107 @@ const ProcessorDetailPage = (): JSX.Element => {
+         */
+        history.replace("/processor-instance-not-found", {
+          title: t("instance.notFound"),
+          message: t("processor.errors.cantFindInstance"),
+        });
+      } else {
+        throw new ErrorWithDetail(
+          (
+            <TextContent>
+              <Text component="h1">
+                {processor?.name ?? t("common.processor")}
+              </Text>
+            </TextContent>
+          ),
+          t("processor.errors.instanceDetailsGenericError")
+        );
+      }
+    }
+  }, [bridgeError, history, processor?.name, t]);
+
+  useEffect(() => {
+    if (processorError && axios.isAxiosError(processorError)) {
+      if (
+        isServiceApiError(processorError) &&
+        getErrorCode(processorError) === APIErrorCodes.ERROR_4
+      ) {
+        /* When the instance is not found on the server, we are going to replace
+         * the current URL with a fake URL that does not match any route.
+         * In this way, the PageNotFound component will be shown.
+         */
+        history.replace("/processor-not-found", {
+          title: t("processor.notFound"),
+          message: t("processor.errors.cantFindProcessor"),
+        });
+      } else {
+        throw new ErrorWithDetail(
+          (
+            <TextContent>
+              <Text component="h1">
+                {processor?.name ?? t("common.processor")}
+              </Text>
+            </TextContent>
+          ),
+          t("processor.errors.processorDetailsGenericError")
+        );
+      }
     }
-    if (processorError) {
-      console.error(processorError);
-      goToInstance();
+  }, [history, processor?.name, processorError, t]);
+
+  useEffect(() => {
+    if (schemasError && axios.isAxiosError(schemasError)) {
+      throw new ErrorWithDetail(
+        (
+          <TextContent>
+            <Text component="h1">
+              {processor?.name ?? t("common.processor")}
+            </Text>
+          </TextContent>
+        ),
+        t("processor.errors.processorDetailsGenericError")
+      );
     }
+  }, [processor?.name, schemasError, t]);
+
+  useEffect(() => {
     if (updateProcessorError && axios.isAxiosError(updateProcessorError)) {
-      // TODO: replace error code string with a value coming from an error catalog
-      //  See https://issues.redhat.com/browse/MGDOBR-669 for more details.
       if (
-        (updateProcessorError.response?.data as ResponseError).code ===
-        "OPENBRIDGE-1"
+        isServiceApiError(updateProcessorError) &&
+        getErrorCode(updateProcessorError) === APIErrorCodes.ERROR_1
       ) {
         setExistingProcessorName(requestData?.name);
+      } else if (
+        isServiceApiError(updateProcessorError) &&
+        getErrorCode(updateProcessorError) === APIErrorCodes.ERROR_19
+      ) {
+        setShowActionModal(true);
+        actionModalFn.current = (): void => {
+          setShowActionModal(false);
+          history.push("/temp");
```

Comment:
I understand the other _magic_ URLs to show the "Not Found" page.

What voodoo does `/temp` do for us? Is it worthy of a code comment?

Reply:
It is a sort of limitation by React Router. In order to refresh the component state, without reloading the page, you need to redirect to another page (we used `/temp` in this case) and then come back to the original one. 
This is also the reason why I proposed https://docs.google.com/document/d/1-j79KIEGu_d8pNYVODIaNbi2rjxxVIgZ4m_OFQwZAeQ/edit?disco=AAAAcM4GK-s
Adding a comment.

Reply:
I don't know if it's because my browser is really slow these days, but I can see a flash of the "not found" page before the reload happening with the `history.goBack()` on next line.

I would agree to your proposal of redirecting to the bridge detail for now.